uefi-test-runner/src/main.rs

```diff
@@ -25,6 +25,12 @@ fn efi_main(image: Handle, mut st: SystemTable<Boot>) -> Status {
     // Initialize utilities (logging, memory allocation...)
     uefi_services::init(&mut st).expect_success("Failed to initialize utilities");
 
+    /* unit tests here */
+    /*let mut buf = String::new();
+    st.firmware_vendor().as_str_in_buf(&mut buf).unwrap();
+    st.stdout().write_str(buf.as_str()).unwrap();
+    assert_eq!("EDK II", buf.as_str());*/
```

Comment:
The unit test seems pretty good. I guess that instead of asserting the value of the firmware vendor (which might change on non-QEMU platforms), we could instead just print it (something like `info!("Firmware vendor: {}", ...)`)

Reply:
Alternatively, you could manually create a `CStr16` from an array of `u16`s and assert the values of `as_str_in_buf`/`as_string`.

Reply:
> The unit test seems pretty good. I guess that instead of asserting the value of the firmware vendor (which might change on non-QEMU platforms), we could instead just print it (something like `info!("Firmware vendor: {}", ...)`)

I'd love to know what you think about https://github.com/rust-osdev/uefi-rs/pull/249/commits/052c6cb356face38812dd780ce11fcb89fe24a0c

This might be a practival solution for other parts of the code/the tests too.

Reply:
That would definitely allow us to add an assertion, but would tie us to OVMF. If they decide to change the vendor string to `EDK III`, the test would needlessly fail :sweat_smile:

Reply:
yes, of course. I just wanted to give you a pointer that the `runs_inside_qemu`-crate might be useful. I reverted the commit for this PR. Are we now good to go?